vm/src/pyobject.rs

```diff
@@ -735,11 +650,20 @@ impl Default for PyContext {
 /// This is an actual python object. It consists of a `typ` which is the
 /// python class, and carries some rust payload optionally. This rust
 /// payload can be a rust float or rust int in case of float and int objects.
-#[derive(Default)]
 pub struct PyObject {
-    pub payload: PyObjectPayload,
     pub typ: Option<PyObjectRef>,
     pub dict: Option<RefCell<PyAttributes>>, // __dict__ member
+    pub payload: Box<dyn Any>,
+}
+
+impl Default for PyObject {
```

Comment:
in what circumstances do we need a default PyObject?

Reply:
This was already there, it just can't be automatically derived anymore. It's used in the `_nothing()` method that it used for the dict, object, and type types. There's a lot of weirdness there that'll need some work.

Reply:
This was already there, it just can't be automatically derived anymore. It's used in the _nothing() method that it used for the dict, object, and type types. There's a lot of weirdness there that'll need some work.

Reply:
I introduced it for defaulting of the new attributes method, and got to simplify _nothing as a side benefit. (I'm not sure I ended up using it for attributes in the end.) It's probably easy to drop Default and just put the default value in _nothing.